framework-cloud/framework-openshift/src/test/java/org/kie/cloud/openshift/log/InstancesLogCollectorRunnableTest.java

```diff
@@ -0,0 +1,192 @@
+package org.kie.cloud.openshift.log;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.stream.Collectors;
+
+import org.apache.commons.io.FileUtils;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.kie.cloud.api.deployment.Instance;
+import org.kie.cloud.openshift.deployment.OpenShiftInstance;
+import org.kie.cloud.openshift.resource.Project;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.runners.MockitoJUnitRunner;
+import rx.Observable;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(MockitoJUnitRunner.class)
+public class InstancesLogCollectorRunnableTest {
+
+    private static final String LOG_FOLDER_NAME = "LOG_FOLDER_NAME";
+    private static final String PROJECT_NAME = "PROJECT_NAME";
+    private static final String LOG_OUTPUT_DIRECTORY = "instances";
+    private static final String LOG_SUFFIX = ".log";
+
+    private static final Integer WAIT_BEFORE_MSG_IN_MS = 2000;
```

Comment:
As we are mocking out the OpenShift, what's the point in waiting 2 seconds for each test? The mock can return immediately, unless you want to specifically test the case when it takes some time, which is e.g. the `killBeforeFinished`.

Reply:
I refactored a bit the code for that.
manyInstanceRunning needs also a small tempo to be sure checks are passing ...